cmake/HalideGeneratorHelpers.cmake

```diff
@@ -148,15 +148,24 @@ function(add_halide_library TARGET)
         message(FATAL_ERROR "Missing FROM argument specifying a Halide generator target")
     endif ()
 
-    if (NOT TARGET ${ARG_FROM})
-        # FROM is usually an unqualified name; if we are crosscompiling, we might need a
-        # fully-qualified name, so add the default package name and retry
-        set(FQ_ARG_FROM "${PROJECT_NAME}::halide_generators::${ARG_FROM}")
-        if (NOT TARGET ${FQ_ARG_FROM})
-            message(FATAL_ERROR "Unable to locate FROM as either ${ARG_FROM} or ${FQ_ARG_FROM}")
-        endif ()
-        set(ARG_FROM "${FQ_ARG_FROM}")
-    endif ()
+    if(ARG_FROM MATCHES ".py$")
+        make_shell_path(PYTHONPATH "$<TARGET_FILE_DIR:Halide::Python>")
```

Comment:
I'm not sure if we're installing our `make_shell_path` helper. If not, this will be broken after install. It's probably broken anyway if this module doesn't `include(MakeShellPath)`.

Reply:
I think we should add an integration test here that makes sure the installed package works for a simple Python generator.

Reply:
> our make_shell_path helper

...I didn't realize that was a helper, I assumed it was a CMake builtin :-/

> an integration test

not sure I understand; how is that different from what we're already doing in the Python tests?

Reply:
> I assumed it was a CMake builtin :-/

IIRC it works around a bug in `$<SHELL_PATH>` that was fixed in 3.18 or something.

Reply:
> not sure I understand; how is that different from what we're already doing in the Python tests?

The Python tests are run in-tree, not the way the apps work, where they `find_package(Halide)`.